pkg/receive/handler.go

```diff
@@ -482,6 +564,64 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 	h.writeSamplesTotal.WithLabelValues(strconv.Itoa(responseStatusCode), tenant).Observe(float64(totalSamples))
 }
 
+// QueryMetaMonitoring queries any Prometheus Query API compatible meta-monitoring
+// solution with the configured query for getting current active (head) series of all tenants.
+// It then populates tenantCurrentSeries map with result.
+func (a *activeSeriesLimit) QueryMetaMonitoring(ctx context.Context, logger log.Logger) error {
+	a.mtx.Lock()
+	defer a.mtx.Unlock()
```

Comment:
Perhaps we don't need this whole section locked and we could just lock / unlock when writing to map? (same with reading with the other method)